contracts/exchangeIssuance/ExchangeIssuance.sol

```diff
@@ -0,0 +1,244 @@
+        _router.swapTokensForExactTokens(_amount, PreciseUnitMath.maxUint256(), path, address(this), block.timestamp);
+    }
+
+    function purchaseTokenBestPrice(address _token, uint256 _amount) private {
+        uint256 uniPrice = tokenAvailable(uniFactory, _token) ? getBuyPrice(true, _token, _amount) : PreciseUnitMath.maxUint256();
+        uint256 sushiPrice = tokenAvailable(sushiFactory, _token) ? getBuyPrice(false, _token, _amount) : PreciseUnitMath.maxUint256();
+        if (uniPrice <= sushiPrice) {
+            purchaseToken(uniRouter, _token, _amount);
+        } else {
+            purchaseToken(sushiRouter, _token, _amount);
+        }
+    }
+
+    function purchaseTokenExactEther(address _token, uint256 _amount) private {
+        address[] memory path = new address[](2);
+        path[0] = WETH;
+        path[1] = _token;
+
+        uint256 uniAmountOut = tokenAvailable(uniFactory, _token) ? uniRouter.getAmountsOut(_amount, path)[1] : 0;
+        uint256 sushiAmountOut = tokenAvailable(sushiFactory, _token) ? sushiRouter.getAmountsOut(_amount, path)[1] : 0;
+        IUniswapV2Router02 router = uniAmountOut >= sushiAmountOut ? uniRouter : sushiRouter;
+
+        router.swapExactETHForTokens{value: _amount}(0, path, msg.sender, block.timestamp);
+    }
+
+    function sellToken(IUniswapV2Router02 _router, address _token) private {
+        uint256 tokenBalance = IERC20(_token).balanceOf(address(this));
+        address[] memory path = new address[](2);
+        path[0] = _token;
+        path[1] = WETH;
+        _router.swapExactTokensForETH(tokenBalance, 0, path, address(this), block.timestamp);
+    }
+
+    function sellTokenBestPrice(address _token) private {
+        uint256 tokenBalance = IERC20(_token).balanceOf(address(this));
+        uint256 uniPrice = tokenAvailable(uniFactory, _token) ? getSellPrice(true, _token, tokenBalance) : 0;
+        uint256 sushiPrice = tokenAvailable(sushiFactory, _token) ? getSellPrice(false, _token, tokenBalance) : 0;
+        if (uniPrice >= sushiPrice) {
+            sellToken(uniRouter, _token);
+        } else {
+            sellToken(sushiRouter, _token);
+        }
+    }
+
+    function purchaseEtherExactTokens(address _token, uint256 _amount) private {
+        address[] memory path = new address[](2);
+        path[0] = _token;
+        path[1] = WETH;
+
+        uint256 uniAmountOut = tokenAvailable(uniFactory, _token) ? uniRouter.getAmountsOut(_amount, path)[1] : 0;
+        uint256 sushiAmountOut = tokenAvailable(sushiFactory, _token) ? sushiRouter.getAmountsOut(_amount, path)[1] : 0;
+        IUniswapV2Router02 router = uniAmountOut >= sushiAmountOut ? uniRouter : sushiRouter;
+
+        IERC20(_token).approve(address(router), PreciseUnitMath.maxUint256());
+        router.swapExactTokensForETH(_amount, 0, path, address(this), block.timestamp);
+    }
+
+    function getTokensNeeded(address _index, uint256 _amount) private view returns (address[] memory, uint256[] memory) {
```

Comment:
For some of these internal functions that have multiple params and return variables, could you add javadocs so its easier to understand what the return is supposed to be?